gemcalibration/src/common/CalibrationWeb.cc

```diff
@@ -0,0 +1,310 @@
+        <<"store_actionURL(\""<<updateLink<<"\");"<<std::endl
+        <<"</script>"<<std::endl;
+    
+
+    
+}
+
+void gem::calib::CalibrationWeb::calibrationPage(xgi::Input* in, xgi::Output* out)
+{
+    CMSGEMOS_DEBUG("CalibrationWeb::calibrationPage");
+    *out << "<div class=\"xdaq-tab-wrapper\">" << std::endl;
+    *out << "    <div align=\"center\">" << std::endl;
+    *out << "        <form id=\"cal_select\">"<<std::endl;
+    *out << "            <div class=\"form-group row\">" << std::endl;
+    *out << "                <h2><span class=\"label label-info col-md-3\" id=\"cal_scurve\">Select calibration type:" << "</span></h2>" << std::endl;
+    *out << "                <div class=\"col-md-3\">" << std::endl;
+    *out << "                    <select class=\"form-control form-control-lg\" id=\"cal_type_select\" name=\"cal_type_select\" onChange=\"selectCalType()\">" << std::endl;
+    *out << "                        <option disabled selected value> -- select an option -- </option>" << std::endl;
+    *out << "                        <option>GBT Phase Scan</option>" << std::endl;
+    *out << "                        <option>Latency Scan</option>" << std::endl;
+    *out << "                        <option>S-curve Scan</option>" << std::endl;
+    *out << "                        <option>S-bit ARM DAC Scan</option>" << std::endl;
+    *out << "                        <option>ARM DAC Scan</option>" << std::endl;
+    *out << "                        <option>Derive DAC Trim Registers</option>" << std::endl;
+    *out << "                        <option>DAC Scan on VFAT3</option>" << std::endl;
+    *out << "                        <option>Calibrate CFG_THR_ARM_DAC</option>" << std::endl;
+    *out << "                        <option disabled value>Whatever else...</option>" << std::endl;
+    *out << "                    </select>" << std::endl;
+    *out << "                </div>" << std::endl; //<div class=\"col-md-6\">
+    *out << "            </div>" << std::endl; //<div class=\"form-group row\">
+    *out << "        </form>"<< std::endl;
+    *out << "    </div>" << std::endl;
+    *out << "</div>" << std::endl;
+
+    *out << "<div class=\"container col-md-12\" id=\"cal_interface\">" << std::endl;
+    *out << "</div>" << std::endl;
+    
+  
+} 
+
+void gem::calib::CalibrationWeb::applicationPage(xgi::Input* in, xgi::Output* out)
+{
+    CMSGEMOS_DEBUG("CalibrationWeb::applicationPage : Do nothing for the moment, will be eventually filled later");
+} ///CG : remove?
+
+
+void gem::calib::CalibrationWeb::genericRadioSelector( std::string paramName, gem::calib::Calibration::scanParamsRadioSelector radio_param  ,xgi::Output* out)
+{
+    
+    *out << "<h2><span class=\"label label-warning col-md-6\">"<< radio_param.label << "</span></h2>"<< std::endl;
+    *out << "<div class=\"col-md-6\">"<< std::endl;
+    *out << "    <div class=\"form-check\">" << std::endl;
+
+    for (unsigned int i=0; i< radio_param.options.size();++i) {
```

Comment:
Can be range-based?
Also a candidate to be shifted to `JS` codebase or at least employ templated framework a-la https://github.com/monsieurgustav/NLTemplate. Same relates to the rest of the file...